core/server/services/settings/index.js

```diff
@@ -4,16 +4,78 @@
  */
 
 var SettingsModel = require('../../models/settings').Settings,
-    SettingsCache = require('./cache');
+    SettingsCache = require('./cache'),
+    SettingsLoader = require('./loader'),
+    _ = require('lodash'),
+    common = require('../../lib/common');
+    // SettingsFiles = require('./ensure-settings');
 
 module.exports = {
     init: function init() {
+        // TODO: the ensure settings fn up before the settings cache
+        // return SettingsFiles(this.knownSettings()).then(() => {
+
         // Update the defaults
         return SettingsModel.populateDefaults()
             .then(function (settingsCollection) {
                 // Initialise the cache with the result
                 // This will bind to events for further updates
                 SettingsCache.init(settingsCollection);
             });
+        // });
+    },
+
+    // TODO: is this the right place for it? We need a global place where
+    // we can add additional files that are supported.
+    // This can also be a config value for example.
+    knownSettings: function knownSettings() {
+        return ['routes'];
+    },
+
+    /**
+     * Getter for YAML settings objects.
+     * Example: `settingsSrvc.get('routes').then(...)`
+     * @param {String} setting type of setting. Needs to be supported.
+     *                         see `knownSettings`
+     * @returns {Promise<Object>} settingsFile
+     * @description Returns settings object as defined per YAML files in
+     * `/content/settings` directory.
+     */
+    get: function get(setting) {
+        const knownSettings = this.knownSettings();
+
+        if (!setting || _.indexOf(knownSettings, setting) < 0) {
+            return Promise.reject(new common.errors.GhostError({
```

Comment:
I would suggest: `IncorrectUsageError`

Reply:
Ha! Had this before! Just switched back, as I thought I have to change them anyway...